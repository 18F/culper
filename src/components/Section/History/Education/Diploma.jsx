import React from 'react'
import { i18n } from '../../../../config'
import { ValidationElement, DateControl, Text, RadioGroup, Radio, Field, Show } from '../../../Form'

export class DiplomaItem extends ValidationElement {
  constructor (props) {
    super(props)

    this.update = this.update.bind(this)
    this.updateDiploma = this.updateDiploma.bind(this)
    this.updateDiplomaOther = this.updateDiplomaOther.bind(this)
    this.updateDate = this.updateDate.bind(this)
  }

  update (queue) {
    this.props.onUpdate({
      Diploma: this.props.Diploma,
      DiplomaOther: this.props.DiplomaOther,
      Date: this.props.Date,
      ...queue
    })
  }

  updateDiploma (values) {
    this.update({
      Diploma: values
    })
  }

  updateDiplomaOther (values) {
    this.update({
      DiplomaOther: values
    })
  }

  updateDate (values) {
    this.update({
      Date: values
    })
  }

  render () {
    return (
      <div>
        <Field title={i18n.t('history.education.heading.diploma')}
               help="history.education.help.diploma"
               adjustFor="big-buttons"
               shrink={true}
               scrollIntoView={this.props.scrollIntoView}>
          <RadioGroup className="diploma option-list"
                      required={this.props.required}
                      onError={this.props.onError}
                      selectedValue={(this.props.Diploma || {}).value}>
            <Radio name="diploma-highschool"
                   className="diploma-highschool"
                   label={i18n.m('history.education.label.diploma.highschool')}
                   value="High School Diploma"
                   onUpdate={this.updateDiploma}
                   onError={this.props.onError}
                   />
            <Radio name="diploma-associate"
                   className="diploma-associate"
                   label={i18n.m('history.education.label.diploma.associate')}
                   value="Associate"
                   onUpdate={this.updateDiploma}
                   onError={this.props.onError}
                   />
            <Radio name="diploma-bachelor"
                   className="diploma-bachelor"
                   label={i18n.m('history.education.label.diploma.bachelor')}
                   value="Bachelor"
                   onUpdate={this.updateDiploma}
                   onError={this.props.onError}
                   />
            <Radio name="diploma-master"
                   className="diploma-master"
                   label={i18n.m('history.education.label.diploma.master')}
                   value="Master"
                   onUpdate={this.updateDiploma}
                   onError={this.props.onError}
                   />
            <Radio name="diploma-doctorate"
                   className="diploma-doctorate"
                   label={i18n.m('history.education.label.diploma.doctorate')}
                   value="Doctorate"
                   onUpdate={this.updateDiploma}
                   onError={this.props.onError}
                   />
            <Radio name="diploma-professional"
                   className="diploma-professional"
                   label={i18n.m('history.education.label.diploma.professional')}
                   value="Professional"
                   onUpdate={this.updateDiploma}
                   onError={this.props.onError}
                   />
            <Radio name="diploma-other"
                   className="diploma-other"
                   label={i18n.m('history.education.label.diploma.other')}
                   value="Other"
                   onUpdate={this.updateDiploma}
                   onError={this.props.onError}
                   />
          </RadioGroup>
          <Show when={(this.props.Diploma || {}).value === 'Other'}>
            <Field title={i18n.t('history.education.label.diploma.other')}
                   titleSize="label"
                   adjustFor="text"
                   scrollIntoView={this.props.scrollIntoView}>
              <Text name="DiplomaOther"
                    {...this.props.DiplomaOther}
                    className="other"
                    maxlength="100"
                    onUpdate={this.updateDiplomaOther}
                    onError={this.props.onError}
                    required={this.props.required}
                    />
            </Field>
          </Show>
        </Field>

        <Field title={i18n.t('history.education.heading.date')}
               help="history.education.help.date"
               adjustFor="label"
               shrink={true}
               scrollIntoView={this.props.scrollIntoView}>
          <DateControl name="Date"
                       {...this.props.Date}
                       className="date-awarded"
                       hideDay={true}
                       onUpdate={this.updateDate}
                       onError={this.props.onError}
                       required={this.props.required}
                       />
        </Field>
      </div>
    )
  }
}

DiplomaItem.defaultProps = {
  Diploma: {},
  DiplomaOther: {},
  Date: {},
  onUpdate: (queue) => {},
  onError: (value, arr) => { return arr }
}
