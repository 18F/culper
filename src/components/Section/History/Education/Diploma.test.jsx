import React from 'react'
import { shallow, mount } from 'enzyme'
import configureMockStore from 'redux-mock-store'
import { Provider } from 'react-redux'
import { DiplomaItem } from './Diploma'

describe('The DiplomaItem component', () => {
  it('renders without errors', () => {
    const component = shallow(
      <DiplomaItem />
    )

    expect(component.exists()).toBe(true)
    expect(component).toMatchSnapshot()
  })

  it('renders the diploma type field', () => {
    const component = shallow(
      <DiplomaItem />
    )

    expect(component.find('.diploma').length).toEqual(1)
  })

  const mockStore = configureMockStore()

  const mountComponentWithStore = (props = {}, defaultState = {}) => {
    const store = mockStore({ ...defaultState })

    return mount(
      <Provider store={store}>
        <DiplomaItem {...props} />
      </Provider>
    )
  }

  it('implements an onUpdate handler', () => {
    const onUpdate = jest.fn()
    const testProps = {
      Diploma: { value: 'Other' },
      onUpdate,
    }

    const component = mountComponentWithStore(testProps)

    component.find('.diploma-highschool input').simulate('change')
    component.find('.diploma-other input').simulate('change')
    component
      .find('.other input')
      .simulate('change', { target: { name: 'DiplomaOther', value: 'Other' } })
    component
      .find('.date-awarded .month input')
      .simulate('change', { target: { name: 'month', value: '1' } })
    component
      .find('.date-awarded .year input')
      .simulate('change', { target: { name: 'year', value: '2010' } })

    expect(onUpdate.mock.calls.length).toEqual(5)
  })

  describe('with "Other" selected', () => {
    const testProps = {
      Diploma: { value: 'Other' },
    }

    const component = mountComponentWithStore(testProps)

    it('displays a textbox', () => {
      expect(component.find('.other').length).toEqual(1)
    })
  })

  describe('with "Other" not selected', () => {
    const testProps = {
      Diploma: { value: 'Doctorate' },
    }

    const component = mountComponentWithStore(testProps)

    it('does not display a textbox', () => {
      expect(component.find('.other').length).toEqual(0)
    })
  })
})
