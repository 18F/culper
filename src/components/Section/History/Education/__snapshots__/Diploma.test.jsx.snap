// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The DiplomaItem component renders without errors 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <Field
      adjustFor="big-buttons"
      className=""
      comments={false}
      commentsActive={false}
      commentsAdd="comments.add"
      commentsName="Comments"
      commentsRemove="comments.remove"
      commentsRequired={false}
      commentsValue={Object {}}
      errorPrefix=""
      errors={Array []}
      filterErrors={[Function]}
      help="history.education.help.diploma"
      helpActive={false}
      optional={false}
      optionalText=""
      scrollIntoView={true}
      shrink={true}
      title="Provide type of degree(s)/diploma(s) received"
      titleSize="h4"
      validate={true}
    >
      <RadioGroup
        className="diploma option-list option-list-vertical"
        disabled={false}
        onError={[Function]}
        required={false}
      >
        <Radio
          ariaLabel=""
          checked={false}
          className="diploma-highschool"
          disabled={false}
          error={false}
          focus={false}
          label={
            <ReactMarkdown
              containerTagName="div"
              parserOptions={Object {}}
              renderers={
                Object {
                  "link": [Function],
                }
              }
              source="High School Diploma"
            />
          }
          name="diploma-highschool"
          native={false}
          onError={[Function]}
          onUpdate={[Function]}
          valid={false}
          value="High School Diploma"
          valued=""
        />
        <Radio
          ariaLabel=""
          checked={false}
          className="diploma-associate"
          disabled={false}
          error={false}
          focus={false}
          label={
            <ReactMarkdown
              containerTagName="div"
              parserOptions={Object {}}
              renderers={
                Object {
                  "link": [Function],
                }
              }
              source="Associate's"
            />
          }
          name="diploma-associate"
          native={false}
          onError={[Function]}
          onUpdate={[Function]}
          valid={false}
          value="Associate"
          valued=""
        />
        <Radio
          ariaLabel=""
          checked={false}
          className="diploma-bachelor"
          disabled={false}
          error={false}
          focus={false}
          label={
            <ReactMarkdown
              containerTagName="div"
              parserOptions={Object {}}
              renderers={
                Object {
                  "link": [Function],
                }
              }
              source="Bachelor's"
            />
          }
          name="diploma-bachelor"
          native={false}
          onError={[Function]}
          onUpdate={[Function]}
          valid={false}
          value="Bachelor"
          valued=""
        />
        <Radio
          ariaLabel=""
          checked={false}
          className="diploma-master"
          disabled={false}
          error={false}
          focus={false}
          label={
            <ReactMarkdown
              containerTagName="div"
              parserOptions={Object {}}
              renderers={
                Object {
                  "link": [Function],
                }
              }
              source="Master's"
            />
          }
          name="diploma-master"
          native={false}
          onError={[Function]}
          onUpdate={[Function]}
          valid={false}
          value="Master"
          valued=""
        />
        <Radio
          ariaLabel=""
          checked={false}
          className="diploma-doctorate"
          disabled={false}
          error={false}
          focus={false}
          label={
            <ReactMarkdown
              containerTagName="div"
              parserOptions={Object {}}
              renderers={
                Object {
                  "link": [Function],
                }
              }
              source="Doctorate"
            />
          }
          name="diploma-doctorate"
          native={false}
          onError={[Function]}
          onUpdate={[Function]}
          valid={false}
          value="Doctorate"
          valued=""
        />
        <Radio
          ariaLabel=""
          checked={false}
          className="diploma-professional"
          disabled={false}
          error={false}
          focus={false}
          label={
            <ReactMarkdown
              containerTagName="div"
              parserOptions={Object {}}
              renderers={
                Object {
                  "link": [Function],
                }
              }
              source="Professional Degree (eg. MD, DVM, JD)"
            />
          }
          name="diploma-professional"
          native={false}
          onError={[Function]}
          onUpdate={[Function]}
          valid={false}
          value="Professional"
          valued=""
        />
        <Radio
          ariaLabel=""
          checked={false}
          className="diploma-other"
          disabled={false}
          error={false}
          focus={false}
          label={
            <ReactMarkdown
              containerTagName="div"
              parserOptions={Object {}}
              renderers={
                Object {
                  "link": [Function],
                }
              }
              source="Other"
            />
          }
          name="diploma-other"
          native={false}
          onError={[Function]}
          onUpdate={[Function]}
          valid={false}
          value="Other"
          valued=""
        />
      </RadioGroup>
      <Show
        when={false}
      >
        <Field
          adjustFor="text"
          className=""
          comments={false}
          commentsActive={false}
          commentsAdd="comments.add"
          commentsName="Comments"
          commentsRemove="comments.remove"
          commentsRequired={false}
          commentsValue={Object {}}
          errorPrefix=""
          errors={Array []}
          filterErrors={[Function]}
          help=""
          helpActive={false}
          optional={false}
          optionalText=""
          scrollIntoView={true}
          shrink={false}
          title="Other"
          titleSize="label"
          validate={true}
        >
          <Text
            className="other"
            maxlength="100"
            name="DiplomaOther"
            onError={[Function]}
            onUpdate={[Function]}
            prefilter={[Function]}
            prefix=""
            required={false}
            status={true}
            value=""
          />
        </Field>
      </Show>
    </Field>
    <Field
      adjustFor="label"
      className=""
      comments={false}
      commentsActive={false}
      commentsAdd="comments.add"
      commentsName="Comments"
      commentsRemove="comments.remove"
      commentsRequired={false}
      commentsValue={Object {}}
      errorPrefix=""
      errors={Array []}
      filterErrors={[Function]}
      help="history.education.help.date"
      helpActive={false}
      optional={false}
      optionalText=""
      scrollIntoView={true}
      shrink={true}
      title="Date awarded"
      titleSize="h4"
      validate={true}
    >
      <Connect(DateControl)
        className="date-awarded"
        hideDay={true}
        minDateEqualTo={true}
        name="Date"
        onError={[Function]}
        onUpdate={[Function]}
      />
    </Field>
  </div>,
  "nodes": Array [
    <div>
      <Field
        adjustFor="big-buttons"
        className=""
        comments={false}
        commentsActive={false}
        commentsAdd="comments.add"
        commentsName="Comments"
        commentsRemove="comments.remove"
        commentsRequired={false}
        commentsValue={Object {}}
        errorPrefix=""
        errors={Array []}
        filterErrors={[Function]}
        help="history.education.help.diploma"
        helpActive={false}
        optional={false}
        optionalText=""
        scrollIntoView={true}
        shrink={true}
        title="Provide type of degree(s)/diploma(s) received"
        titleSize="h4"
        validate={true}
      >
        <RadioGroup
          className="diploma option-list option-list-vertical"
          disabled={false}
          onError={[Function]}
          required={false}
        >
          <Radio
            ariaLabel=""
            checked={false}
            className="diploma-highschool"
            disabled={false}
            error={false}
            focus={false}
            label={
              <ReactMarkdown
                containerTagName="div"
                parserOptions={Object {}}
                renderers={
                  Object {
                    "link": [Function],
                  }
                }
                source="High School Diploma"
              />
            }
            name="diploma-highschool"
            native={false}
            onError={[Function]}
            onUpdate={[Function]}
            valid={false}
            value="High School Diploma"
            valued=""
          />
          <Radio
            ariaLabel=""
            checked={false}
            className="diploma-associate"
            disabled={false}
            error={false}
            focus={false}
            label={
              <ReactMarkdown
                containerTagName="div"
                parserOptions={Object {}}
                renderers={
                  Object {
                    "link": [Function],
                  }
                }
                source="Associate's"
              />
            }
            name="diploma-associate"
            native={false}
            onError={[Function]}
            onUpdate={[Function]}
            valid={false}
            value="Associate"
            valued=""
          />
          <Radio
            ariaLabel=""
            checked={false}
            className="diploma-bachelor"
            disabled={false}
            error={false}
            focus={false}
            label={
              <ReactMarkdown
                containerTagName="div"
                parserOptions={Object {}}
                renderers={
                  Object {
                    "link": [Function],
                  }
                }
                source="Bachelor's"
              />
            }
            name="diploma-bachelor"
            native={false}
            onError={[Function]}
            onUpdate={[Function]}
            valid={false}
            value="Bachelor"
            valued=""
          />
          <Radio
            ariaLabel=""
            checked={false}
            className="diploma-master"
            disabled={false}
            error={false}
            focus={false}
            label={
              <ReactMarkdown
                containerTagName="div"
                parserOptions={Object {}}
                renderers={
                  Object {
                    "link": [Function],
                  }
                }
                source="Master's"
              />
            }
            name="diploma-master"
            native={false}
            onError={[Function]}
            onUpdate={[Function]}
            valid={false}
            value="Master"
            valued=""
          />
          <Radio
            ariaLabel=""
            checked={false}
            className="diploma-doctorate"
            disabled={false}
            error={false}
            focus={false}
            label={
              <ReactMarkdown
                containerTagName="div"
                parserOptions={Object {}}
                renderers={
                  Object {
                    "link": [Function],
                  }
                }
                source="Doctorate"
              />
            }
            name="diploma-doctorate"
            native={false}
            onError={[Function]}
            onUpdate={[Function]}
            valid={false}
            value="Doctorate"
            valued=""
          />
          <Radio
            ariaLabel=""
            checked={false}
            className="diploma-professional"
            disabled={false}
            error={false}
            focus={false}
            label={
              <ReactMarkdown
                containerTagName="div"
                parserOptions={Object {}}
                renderers={
                  Object {
                    "link": [Function],
                  }
                }
                source="Professional Degree (eg. MD, DVM, JD)"
              />
            }
            name="diploma-professional"
            native={false}
            onError={[Function]}
            onUpdate={[Function]}
            valid={false}
            value="Professional"
            valued=""
          />
          <Radio
            ariaLabel=""
            checked={false}
            className="diploma-other"
            disabled={false}
            error={false}
            focus={false}
            label={
              <ReactMarkdown
                containerTagName="div"
                parserOptions={Object {}}
                renderers={
                  Object {
                    "link": [Function],
                  }
                }
                source="Other"
              />
            }
            name="diploma-other"
            native={false}
            onError={[Function]}
            onUpdate={[Function]}
            valid={false}
            value="Other"
            valued=""
          />
        </RadioGroup>
        <Show
          when={false}
        >
          <Field
            adjustFor="text"
            className=""
            comments={false}
            commentsActive={false}
            commentsAdd="comments.add"
            commentsName="Comments"
            commentsRemove="comments.remove"
            commentsRequired={false}
            commentsValue={Object {}}
            errorPrefix=""
            errors={Array []}
            filterErrors={[Function]}
            help=""
            helpActive={false}
            optional={false}
            optionalText=""
            scrollIntoView={true}
            shrink={false}
            title="Other"
            titleSize="label"
            validate={true}
          >
            <Text
              className="other"
              maxlength="100"
              name="DiplomaOther"
              onError={[Function]}
              onUpdate={[Function]}
              prefilter={[Function]}
              prefix=""
              required={false}
              status={true}
              value=""
            />
          </Field>
        </Show>
      </Field>
      <Field
        adjustFor="label"
        className=""
        comments={false}
        commentsActive={false}
        commentsAdd="comments.add"
        commentsName="Comments"
        commentsRemove="comments.remove"
        commentsRequired={false}
        commentsValue={Object {}}
        errorPrefix=""
        errors={Array []}
        filterErrors={[Function]}
        help="history.education.help.date"
        helpActive={false}
        optional={false}
        optionalText=""
        scrollIntoView={true}
        shrink={true}
        title="Date awarded"
        titleSize="h4"
        validate={true}
      >
        <Connect(DateControl)
          className="date-awarded"
          hideDay={true}
          minDateEqualTo={true}
          name="Date"
          onError={[Function]}
          onUpdate={[Function]}
        />
      </Field>
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <DiplomaItem
        Date={Object {}}
        Diploma={Object {}}
        DiplomaOther={Object {}}
        onError={[Function]}
        onUpdate={[Function]}
      />,
      "_debugID": 296,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": DiplomaItem {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "handleBlur": [Function],
        "handleChange": [Function],
        "handleFocus": [Function],
        "handleKeyDown": [Function],
        "handleValidation": [Function],
        "props": Object {
          "Date": Object {},
          "Diploma": Object {},
          "DiplomaOther": Object {},
          "onError": [Function],
          "onUpdate": [Function],
        },
        "refs": Object {},
        "state": null,
        "update": [Function],
        "updateDate": [Function],
        "updateDiploma": [Function],
        "updateDiplomaOther": [Function],
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <Field
            adjustFor="big-buttons"
            className=""
            comments={false}
            commentsActive={false}
            commentsAdd="comments.add"
            commentsName="Comments"
            commentsRemove="comments.remove"
            commentsRequired={false}
            commentsValue={Object {}}
            errorPrefix=""
            errors={Array []}
            filterErrors={[Function]}
            help="history.education.help.diploma"
            helpActive={false}
            optional={false}
            optionalText=""
            scrollIntoView={true}
            shrink={true}
            title="Provide type of degree(s)/diploma(s) received"
            titleSize="h4"
            validate={true}
          >
            <RadioGroup
              className="diploma option-list option-list-vertical"
              disabled={false}
              onError={[Function]}
              required={false}
            >
              <Radio
                ariaLabel=""
                checked={false}
                className="diploma-highschool"
                disabled={false}
                error={false}
                focus={false}
                label={
                  <ReactMarkdown
                    containerTagName="div"
                    parserOptions={Object {}}
                    renderers={
                      Object {
                        "link": [Function],
                      }
                    }
                    source="High School Diploma"
                  />
                }
                name="diploma-highschool"
                native={false}
                onError={[Function]}
                onUpdate={[Function]}
                valid={false}
                value="High School Diploma"
                valued=""
              />
              <Radio
                ariaLabel=""
                checked={false}
                className="diploma-associate"
                disabled={false}
                error={false}
                focus={false}
                label={
                  <ReactMarkdown
                    containerTagName="div"
                    parserOptions={Object {}}
                    renderers={
                      Object {
                        "link": [Function],
                      }
                    }
                    source="Associate's"
                  />
                }
                name="diploma-associate"
                native={false}
                onError={[Function]}
                onUpdate={[Function]}
                valid={false}
                value="Associate"
                valued=""
              />
              <Radio
                ariaLabel=""
                checked={false}
                className="diploma-bachelor"
                disabled={false}
                error={false}
                focus={false}
                label={
                  <ReactMarkdown
                    containerTagName="div"
                    parserOptions={Object {}}
                    renderers={
                      Object {
                        "link": [Function],
                      }
                    }
                    source="Bachelor's"
                  />
                }
                name="diploma-bachelor"
                native={false}
                onError={[Function]}
                onUpdate={[Function]}
                valid={false}
                value="Bachelor"
                valued=""
              />
              <Radio
                ariaLabel=""
                checked={false}
                className="diploma-master"
                disabled={false}
                error={false}
                focus={false}
                label={
                  <ReactMarkdown
                    containerTagName="div"
                    parserOptions={Object {}}
                    renderers={
                      Object {
                        "link": [Function],
                      }
                    }
                    source="Master's"
                  />
                }
                name="diploma-master"
                native={false}
                onError={[Function]}
                onUpdate={[Function]}
                valid={false}
                value="Master"
                valued=""
              />
              <Radio
                ariaLabel=""
                checked={false}
                className="diploma-doctorate"
                disabled={false}
                error={false}
                focus={false}
                label={
                  <ReactMarkdown
                    containerTagName="div"
                    parserOptions={Object {}}
                    renderers={
                      Object {
                        "link": [Function],
                      }
                    }
                    source="Doctorate"
                  />
                }
                name="diploma-doctorate"
                native={false}
                onError={[Function]}
                onUpdate={[Function]}
                valid={false}
                value="Doctorate"
                valued=""
              />
              <Radio
                ariaLabel=""
                checked={false}
                className="diploma-professional"
                disabled={false}
                error={false}
                focus={false}
                label={
                  <ReactMarkdown
                    containerTagName="div"
                    parserOptions={Object {}}
                    renderers={
                      Object {
                        "link": [Function],
                      }
                    }
                    source="Professional Degree (eg. MD, DVM, JD)"
                  />
                }
                name="diploma-professional"
                native={false}
                onError={[Function]}
                onUpdate={[Function]}
                valid={false}
                value="Professional"
                valued=""
              />
              <Radio
                ariaLabel=""
                checked={false}
                className="diploma-other"
                disabled={false}
                error={false}
                focus={false}
                label={
                  <ReactMarkdown
                    containerTagName="div"
                    parserOptions={Object {}}
                    renderers={
                      Object {
                        "link": [Function],
                      }
                    }
                    source="Other"
                  />
                }
                name="diploma-other"
                native={false}
                onError={[Function]}
                onUpdate={[Function]}
                valid={false}
                value="Other"
                valued=""
              />
            </RadioGroup>
            <Show
              when={false}
            >
              <Field
                adjustFor="text"
                className=""
                comments={false}
                commentsActive={false}
                commentsAdd="comments.add"
                commentsName="Comments"
                commentsRemove="comments.remove"
                commentsRequired={false}
                commentsValue={Object {}}
                errorPrefix=""
                errors={Array []}
                filterErrors={[Function]}
                help=""
                helpActive={false}
                optional={false}
                optionalText=""
                scrollIntoView={true}
                shrink={false}
                title="Other"
                titleSize="label"
                validate={true}
              >
                <Text
                  className="other"
                  maxlength="100"
                  name="DiplomaOther"
                  onError={[Function]}
                  onUpdate={[Function]}
                  prefilter={[Function]}
                  prefix=""
                  required={false}
                  status={true}
                  value=""
                />
              </Field>
            </Show>
          </Field>
          <Field
            adjustFor="label"
            className=""
            comments={false}
            commentsActive={false}
            commentsAdd="comments.add"
            commentsName="Comments"
            commentsRemove="comments.remove"
            commentsRequired={false}
            commentsValue={Object {}}
            errorPrefix=""
            errors={Array []}
            filterErrors={[Function]}
            help="history.education.help.date"
            helpActive={false}
            optional={false}
            optionalText=""
            scrollIntoView={true}
            shrink={true}
            title="Date awarded"
            titleSize="h4"
            validate={true}
          >
            <Connect(DateControl)
              className="date-awarded"
              hideDay={true}
              minDateEqualTo={true}
              name="Date"
              onError={[Function]}
              onUpdate={[Function]}
            />
          </Field>
        </div>,
        "_debugID": 297,
        "_renderedOutput": <div>
          <Field
            adjustFor="big-buttons"
            className=""
            comments={false}
            commentsActive={false}
            commentsAdd="comments.add"
            commentsName="Comments"
            commentsRemove="comments.remove"
            commentsRequired={false}
            commentsValue={Object {}}
            errorPrefix=""
            errors={Array []}
            filterErrors={[Function]}
            help="history.education.help.diploma"
            helpActive={false}
            optional={false}
            optionalText=""
            scrollIntoView={true}
            shrink={true}
            title="Provide type of degree(s)/diploma(s) received"
            titleSize="h4"
            validate={true}
          >
            <RadioGroup
              className="diploma option-list option-list-vertical"
              disabled={false}
              onError={[Function]}
              required={false}
            >
              <Radio
                ariaLabel=""
                checked={false}
                className="diploma-highschool"
                disabled={false}
                error={false}
                focus={false}
                label={
                  <ReactMarkdown
                    containerTagName="div"
                    parserOptions={Object {}}
                    renderers={
                      Object {
                        "link": [Function],
                      }
                    }
                    source="High School Diploma"
                  />
                }
                name="diploma-highschool"
                native={false}
                onError={[Function]}
                onUpdate={[Function]}
                valid={false}
                value="High School Diploma"
                valued=""
              />
              <Radio
                ariaLabel=""
                checked={false}
                className="diploma-associate"
                disabled={false}
                error={false}
                focus={false}
                label={
                  <ReactMarkdown
                    containerTagName="div"
                    parserOptions={Object {}}
                    renderers={
                      Object {
                        "link": [Function],
                      }
                    }
                    source="Associate's"
                  />
                }
                name="diploma-associate"
                native={false}
                onError={[Function]}
                onUpdate={[Function]}
                valid={false}
                value="Associate"
                valued=""
              />
              <Radio
                ariaLabel=""
                checked={false}
                className="diploma-bachelor"
                disabled={false}
                error={false}
                focus={false}
                label={
                  <ReactMarkdown
                    containerTagName="div"
                    parserOptions={Object {}}
                    renderers={
                      Object {
                        "link": [Function],
                      }
                    }
                    source="Bachelor's"
                  />
                }
                name="diploma-bachelor"
                native={false}
                onError={[Function]}
                onUpdate={[Function]}
                valid={false}
                value="Bachelor"
                valued=""
              />
              <Radio
                ariaLabel=""
                checked={false}
                className="diploma-master"
                disabled={false}
                error={false}
                focus={false}
                label={
                  <ReactMarkdown
                    containerTagName="div"
                    parserOptions={Object {}}
                    renderers={
                      Object {
                        "link": [Function],
                      }
                    }
                    source="Master's"
                  />
                }
                name="diploma-master"
                native={false}
                onError={[Function]}
                onUpdate={[Function]}
                valid={false}
                value="Master"
                valued=""
              />
              <Radio
                ariaLabel=""
                checked={false}
                className="diploma-doctorate"
                disabled={false}
                error={false}
                focus={false}
                label={
                  <ReactMarkdown
                    containerTagName="div"
                    parserOptions={Object {}}
                    renderers={
                      Object {
                        "link": [Function],
                      }
                    }
                    source="Doctorate"
                  />
                }
                name="diploma-doctorate"
                native={false}
                onError={[Function]}
                onUpdate={[Function]}
                valid={false}
                value="Doctorate"
                valued=""
              />
              <Radio
                ariaLabel=""
                checked={false}
                className="diploma-professional"
                disabled={false}
                error={false}
                focus={false}
                label={
                  <ReactMarkdown
                    containerTagName="div"
                    parserOptions={Object {}}
                    renderers={
                      Object {
                        "link": [Function],
                      }
                    }
                    source="Professional Degree (eg. MD, DVM, JD)"
                  />
                }
                name="diploma-professional"
                native={false}
                onError={[Function]}
                onUpdate={[Function]}
                valid={false}
                value="Professional"
                valued=""
              />
              <Radio
                ariaLabel=""
                checked={false}
                className="diploma-other"
                disabled={false}
                error={false}
                focus={false}
                label={
                  <ReactMarkdown
                    containerTagName="div"
                    parserOptions={Object {}}
                    renderers={
                      Object {
                        "link": [Function],
                      }
                    }
                    source="Other"
                  />
                }
                name="diploma-other"
                native={false}
                onError={[Function]}
                onUpdate={[Function]}
                valid={false}
                value="Other"
                valued=""
              />
            </RadioGroup>
            <Show
              when={false}
            >
              <Field
                adjustFor="text"
                className=""
                comments={false}
                commentsActive={false}
                commentsAdd="comments.add"
                commentsName="Comments"
                commentsRemove="comments.remove"
                commentsRequired={false}
                commentsValue={Object {}}
                errorPrefix=""
                errors={Array []}
                filterErrors={[Function]}
                help=""
                helpActive={false}
                optional={false}
                optionalText=""
                scrollIntoView={true}
                shrink={false}
                title="Other"
                titleSize="label"
                validate={true}
              >
                <Text
                  className="other"
                  maxlength="100"
                  name="DiplomaOther"
                  onError={[Function]}
                  onUpdate={[Function]}
                  prefilter={[Function]}
                  prefix=""
                  required={false}
                  status={true}
                  value=""
                />
              </Field>
            </Show>
          </Field>
          <Field
            adjustFor="label"
            className=""
            comments={false}
            commentsActive={false}
            commentsAdd="comments.add"
            commentsName="Comments"
            commentsRemove="comments.remove"
            commentsRequired={false}
            commentsValue={Object {}}
            errorPrefix=""
            errors={Array []}
            filterErrors={[Function]}
            help="history.education.help.date"
            helpActive={false}
            optional={false}
            optionalText=""
            scrollIntoView={true}
            shrink={true}
            title="Date awarded"
            titleSize="h4"
            validate={true}
          >
            <Connect(DateControl)
              className="date-awarded"
              hideDay={true}
              minDateEqualTo={true}
              name="Date"
              onError={[Function]}
              onUpdate={[Function]}
            />
          </Field>
        </div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <DiplomaItem
    Date={Object {}}
    Diploma={Object {}}
    DiplomaOther={Object {}}
    onError={[Function]}
    onUpdate={[Function]}
  />,
}
`;
